src/camp_core.F90

```diff
@@ -1403,31 +1485,93 @@ subroutine solve(this, camp_state, time_step, rxn_phase, solver_stats)
     ! Make sure the requested solver was loaded
     call assert_msg(730097030, associated(solver), "Invalid solver requested")
 
+    t_initial = real(0.0, kind=dp)
+    t_final = time_step
+
     ! Run the integration
     if (present(solver_stats)) then
-      call solver%solve(camp_state, real(0.0, kind=dp), time_step,          &
-                        solver_stats)
+      call solver%get_solver_stats( solver_stats )
+      solver_status = solver%solve(camp_state, t_initial, t_final, solver_stats)
+      solver_stats%status_code   = solver_status
+      solver_stats%start_time__s = t_initial
+      solver_stats%end_time__s   = t_final
     else
-      call solver%solve(camp_state, real(0.0, kind=dp), time_step)
+      solver_status = solver%solve(camp_state, t_initial, t_final)
+    end if
+
+    if (.not.present(solver_stats)) then
+      call warn_assert_msg(997420005, solver_status.eq.0, "Solver failed")
     end if
 
   end subroutine solve
 
+  subroutine export_solver_state(this)
+    use camp_rxn_data
+    use iso_c_binding
+    class(camp_core_t), intent(inout) :: this
+    integer(kind=i_kind) :: phase
+    type(camp_solver_data_t), pointer :: solver
+    phase = GAS_AERO_RXN
+    if (phase.eq.GAS_RXN) then
+      solver => this%solver_data_gas
+    else if (phase.eq.AERO_RXN) then
+      solver => this%solver_data_aero
+    else if (phase.eq.GAS_AERO_RXN) then
+      solver => this%solver_data_gas_aero
+    end if
+    call solver%export_solver_data_state()
+  end subroutine
+
+  subroutine join_solver_state(this)
+    use camp_rxn_data
+    use iso_c_binding
+    class(camp_core_t), intent(inout) :: this
+    integer(kind=i_kind) :: phase
+    type(camp_solver_data_t), pointer :: solver
+    phase = GAS_AERO_RXN
+    if (phase.eq.GAS_RXN) then
+      solver => this%solver_data_gas
+    else if (phase.eq.AERO_RXN) then
+      solver => this%solver_data_aero
+    else if (phase.eq.GAS_AERO_RXN) then
+      solver => this%solver_data_gas_aero
+    end if
+    call solver%join_solver_data_state()
+  end subroutine
+
+  subroutine export_solver_stats(this)
```

Comment:
could you include a description of what this function does?

Reply:
Added:  ` !> Export execution time of GPU and CPU code to calculate speedups at TestMonarch.py`